### PR TITLE
t poly split

### PR DESCRIPTION
**Pull request by liliatangxy**
Splits t poly commitment into deg < srs size polys to handle high degree t polys

### CODE REVIEW

**Comment by Vimalakirti** [comment]
How's its security?

**Comment by liliatangxy** [comment]
The main differences I'm aware of between this and that in the Plonk paper is the degree limit and that there isn't a fixed number of commitments of the t polynomial. I think these are ok because we are committing to the t commitments before getting the opening.

**Comment by ddkang** [comment]
> The main differences I'm aware of between this and that in the Plonk paper is the degree limit and that there isn't a fixed number of commitments of the t polynomial. I think these are ok because we are committing to the t commitments before getting the opening.

Think is not good enough for security. Please produce a proof sketch

**Comment by liliatangxy** [comment]
The proof for statistical zero knowledge is the same as that in the thesis Ari linked. In their case, they use the srs degree limit as the degree of the split, and they used a fixed number of t poly commitments. In our case, the number of t poly commitments is deterministic depending on the input size and srs size.

Here is the link to the thesis: https://repositum.tuwien.at/bitstream/20.500.12708/188597/1/Sefranek%20Marek%20-%202023%20-%20How%20to%20Simulate%20PLONK%20A%20Formal%20Security%20Analysis%20of%20a...pdf

File: src/util/poly.rs [hunk 1]
```diff
@@ -52,3 +52,7 @@ pub fn mul_polys(polys: &Vec<DensePolynomial<Fr>>) -> DensePolynomial<Fr> {
   // Recursively call mul_polys on the next level until we get the root
   mul_polys(&next_level)
```
Comment (Vimalakirti) [inline src/util/poly.rs:53]: Do we still need #131?

Comment (liliatangxy) [reply src/util/poly.rs:53]: Yes since we still will need to do the full multiplication

Comment (Vimalakirti) [reply src/util/poly.rs:53]: Sure, then what's the proper order to merge them?

Comment (liliatangxy) [reply src/util/poly.rs:53]: They can be merged independently

**Review by Vimalakirti — APPROVED** [review]

**Comment by ddkang** [comment]
Is there any performance degradation for no split?

**Comment by liliatangxy** [comment]
No, tested on a resnet18 convolution

**Review by ddkang — APPROVED** [review]

**Pull request merged**